File: src/main/java/ru/stnk/RestTestAPI/service/UserDetailsImpl.java
package ru.stnk.RestTestAPI.service;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;
import ru.stnk.RestTestAPI.entity.Roles;
import ru.stnk.RestTestAPI.entity.User;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

public class UserDetailsImpl implements UserDetails {

    private User user;

    public UserDetailsImpl (User user) {
        this.user = user;
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {

        Collection<Roles> listRoles = new ArrayList<>();

        List<Roles> userRoles = user.getRoles();

        listRoles.addAll(userRoles);

        return listRoles;
    }

    @Override
    public String getUsername() {
        return user.getEmail();
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return user.isEnabled();
    }

    @Override
    public String getPassword() {
        return user.getPassword();
    }

}

File: src/main/kotlin/ru/stnk/resttestapi/service/UserDetailsImpl.kt
package ru.stnk.resttestapi.service

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import ru.stnk.resttestapi.entity.Roles
import ru.stnk.resttestapi.entity.User
import java.util.*

class UserDetailsImpl(private val user: User) : UserDetails {

    override fun getAuthorities(): Collection<GrantedAuthority> {

        val listRoles = ArrayList<Roles>()

        val userRoles = user.roles

        listRoles.addAll(userRoles)

        return listRoles
    }

    override fun getUsername(): String? {
        return user.email
    }

    override fun isAccountNonExpired(): Boolean {
        return true
    }

    override fun isAccountNonLocked(): Boolean {
        return true
    }

    override fun isCredentialsNonExpired(): Boolean {
        return true
    }

    override fun isEnabled(): Boolean {
        return user.isEnabled
    }

    override fun getPassword(): String? {
        return user.password
    }

}
